src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -294,16 +292,12 @@ def get_models(self, *attr_args, **kwargs):
         return get_sdk(self.cli_ctx, resource_type, *attr_args, mod='models')
 
     def command_group(self, group_name, command_type=None, **kwargs):
-        merged_kwargs = self.module_kwargs.copy()
         if command_type:
-            merged_kwargs['command_type'] = command_type
-        merged_kwargs.update(kwargs)
-        return self._command_group_cls(self, group_name, **merged_kwargs)
+            kwargs['command_type'] = command_type
+        return self._command_group_cls(self, group_name, **kwargs)
 
     def argument_context(self, scope, **kwargs):
-        merged_kwargs = self.module_kwargs.copy()
-        merged_kwargs.update(kwargs)
-        return self._argument_context_cls(self, scope, **merged_kwargs)
+        return self._argument_context_cls(self, scope, **kwargs)
 
     def _cli_command(self, name, operation=None, handler=None, argument_loader=None, description_loader=None, **kwargs):
 
```

Comment:
can you simplify the code dealing with `client_factory` in this method as well?
(lines 313-315)

Reply:
Possibly. I'll take a look.

Reply:
Done please take another look.

Reply:
This changed exposed some issues in ACS and Batch.  I'll take a look. I suspect it is just a matter of fixing the command registrations.